### PR TITLE
Add `requires-python` to `uv init`

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

Prefers, in order:

- The major-minor version of an interpreter discovered via `--python`.
- The `requires-python` from the workspace.
- The major-minor version of the default interpreter.

If the `--python` request is a version or a version range, we use that without fetching an interpreter.

Closes https://github.com/astral-sh/uv/issues/5299.

### CODE REVIEW

File: crates/uv/src/commands/project/init.rs [hunk 1]
```diff
@@ -92,6 +102,90 @@ pub(crate) async fn init(
...
+                    EnvironmentPreference::Any,
+                    python_preference,
+                    python_fetch,
+                    &client_builder,
+                    cache,
+                    Some(&reporter),
+                )
+                .await?
+                .into_interpreter();
+                RequiresPython::greater_than_equal_version(&interpreter.python_minor_version())
```
Comment (charliermarsh) [inline crates/uv/src/commands/project/init.rs:141]: @zanieb - Can you review that what I'm doing with the discovery here is reasonable?

**Review by konstin — APPROVED** [review]

**Comment by charliermarsh** [comment]
Yeah... Though unfortunately we need to create the `pyproject.toml` to discover the workspace :(

I'll figure something out.

**Comment by j178** [comment]
Just a heads-up,`cargo init` checks `workspace.members` glob pattern before creating the project:

https://github.com/rust-lang/cargo/blob/7f2079838ac6e0a3cb05b1b9dac4401a76572855/src/cargo/ops/cargo_new.rs#L978-L996

**Comment by charliermarsh** [comment]
@j178 - We also check that before modifying the workspace manifest.

**Review by ibraheemdev — APPROVED** [review]

**Pull request merged**